subprojects/plugins/src/main/java/org/gradle/api/tasks/bundling/War.java

```diff
@@ -140,7 +140,7 @@ public CopySpec webInf(Action<? super CopySpec> configureAction) {
      *
      * @return The classpath. Returns an empty collection when there is no classpath to include in the WAR.
      */
-    @Optional
+    @Nullable
```

Comment:
Should we annotate setter parameters with `@Nullable` too?

Reply:
We should. The question is if I should do this changes as part of this PR. WDYT?

Reply:
I think if we annotate the getters, we should do the same with the setters, yes.